lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/driver/AzureBatchEvaluatorShimConfigurationProvider.java

```diff
@@ -32,10 +38,23 @@
 public class AzureBatchEvaluatorShimConfigurationProvider {
 
   private final RemoteManager remoteManager;
+  private final TcpPortProvider portProvider;
+  private final LocalAddressProvider localAddressProvider;
+  private final Set<String> tcpPortSet;
 
   @Inject
-  AzureBatchEvaluatorShimConfigurationProvider(final RemoteManager remoteManager) {
+  AzureBatchEvaluatorShimConfigurationProvider(
+      @Parameter(TcpPortSet.class) final Set<Integer> tcpPortSet,
+      final RemoteManager remoteManager,
+      final LocalAddressProvider localAddressProvider,
+      final TcpPortProvider portProvider) {
     this.remoteManager = remoteManager;
+    this.portProvider = portProvider;
+    this.localAddressProvider = localAddressProvider;
+    this.tcpPortSet = new HashSet<String>(tcpPortSet.size());
```

Comment:
> this.tcpPortSet = new HashSet<String>(tcpPortSet.size()) [](start = 3, length = 57)

Why are we converting between string and integer?

Reply:
ConfigurationModule.setMultiple only accepts a set of strings. That is why I had to convert from ints to strings

---
In reply to: [201094585](https://github.com/sharathmalladi/reef/pull/5#discussion_r201094585) [](ancestors = 201094585)

Reply:
Add comment to explain conversion.

---
In reply to: [201473975](https://github.com/sharathmalladi/reef/pull/5#discussion_r201473975) [](ancestors = 201473975,201094585)